src/plugins/intel_gpu/src/plugin/plugin.cpp

```diff
@@ -852,8 +852,30 @@ uint32_t Plugin::get_optimal_batch_size(const ov::AnyMap& options) const {
     return batch;
 }
 
+class PluginSentry {
+public:
+    PluginSentry() {
+#ifdef ENABLE_ONEDNN_FOR_GPU
+        // touch oneDNN mutex for correct static objects destruction
+        auto capacity = dnnl::get_primitive_cache_capacity();
+        dnnl::set_primitive_cache_capacity(0);
+        dnnl::set_primitive_cache_capacity(capacity);
+#endif
+    }
+
+    ~PluginSentry() {
+#ifdef ENABLE_ONEDNN_FOR_GPU
+        // reset the capacity when plugin deleted
+        auto capacity = dnnl::get_primitive_cache_capacity();
+        dnnl::set_primitive_cache_capacity(0);
+        dnnl::set_primitive_cache_capacity(capacity);
+#endif
+    }
+};
+
 }  // namespace intel_gpu
 }  // namespace ov
 
 static const ov::Version version = { CI_BUILD_NUMBER, "Intel GPU plugin" };
+static ov::intel_gpu::PluginSentry sentry;
```

Comment:
is it possible somehow release memory once all `CompiledModel`s are released, but plugin is still alive?

Reply:
For the moment cache is global and not related to model.  The local, model related cache is discussed with oneDNN now